infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -546,92 +583,48 @@ public void updateSyncStatus(String clusterId, SyncType lastSyncType, SyncStatus
     }
 
     /**
-     * Set replication status table.
-     * If the current sync type is log entry sync, keep Snapshot Sync Info.
+     * Updates the number of remaining entries.
      *
      * Note: TransactionAbortedException has been handled by upper level.
      *
      * @param clusterId standby cluster id
      * @param remainingEntries num of remaining entries to send
      * @param type sync type
      */
-    public void setReplicationStatusTable(String clusterId, long remainingEntries, SyncType type) {
-        ReplicationStatusKey key = ReplicationStatusKey.newBuilder().setClusterId(clusterId).build();
-        SnapshotSyncInfo snapshotStatus = null;
-        ReplicationStatusVal current;
-        ReplicationStatusVal previous = null;
-
+    public void updateRemainingEntriesToSend(String clusterId, long remainingEntries, SyncType type) {
         try (TxnContext txn = corfuStore.txn(NAMESPACE)) {
-            CorfuStoreEntry<ReplicationStatusKey, ReplicationStatusVal, Message> record = txn.getRecord(replicationStatusTable, key);
-            if (record.getPayload() != null) {
-                previous = record.getPayload();
-                snapshotStatus = previous.getSnapshotSyncInfo();
-            }
-            txn.commit();
-        }
+            ReplicationStatusKey key = ReplicationStatusKey.newBuilder().setClusterId(clusterId).build();
+            CorfuStoreEntry<ReplicationStatusKey, ReplicationStatusVal, Message> entry =
+                    txn.getRecord(replicationStatusTable, key);
+
+            ReplicationStatusVal previous = entry.getPayload();
+            SnapshotSyncInfo previousSnapshotSyncInfo = previous.getSnapshotSyncInfo();
 
-        if (type == SyncType.LOG_ENTRY) {
-            if (previous != null &&
-                    (previous.getStatus().equals(SyncStatus.NOT_STARTED)
-                            || snapshotStatus.getStatus().equals(SyncStatus.STOPPED))) {
-                log.info("syncStatusPoller :: skip replication status update, log entry replication is {}", previous.getStatus());
+            if (type == SyncType.LOG_ENTRY && (previous.getStatus().equals(SyncStatus.NOT_STARTED)
```

Comment:
Why should we remove the checks for NOT_STARTED? Since that is our indicator that we have not yet proceeded to or in early stages of the next state, we again would introduce a race condition with the polling task. Say for example we exit the initialized state and start the polling task, the first run is fine since the status is still null and we just return, however if the task runs right after we set the status in the next state but before we mark the sync info then the polling task will update the protobuf with the incorrect default value for the sync type.

Reply:
we will not initialize the in-memory lastSyncType before the state transition, so there will be no default value for it.  Lets discuss this also.